server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -99,61 +100,56 @@ private void extensionsDiscovery() throws IOException {
         if (!FileSystemUtils.isAccessibleDirectory(extensionsPath, logger)) {
             return;
         }
-        for (final Path plugin : PluginsService.findPluginDirs(extensionsPath)) {
-            try {
-                PluginInfo pluginInfo = PluginInfo.readFromProperties(plugin);
-
-                // TODO: Add unit tests for commented section below
+        List<Path> pluginDirs = PluginsService.findPluginDirs(extensionsPath);
 
-                /**
+        Set<Extension> extensions = new HashSet<Extension>();
+        if (!pluginDirs.isEmpty()) {
+            try {
+                extensions = readFromExtensionsYml(extensionsPath.resolve("extensions.yml")).getExtensions();
+            } catch (Exception e) {
```

Comment:
This seems overly broad (as does the `throws` clause for `readFrom ExtensionsYml()`